### PR TITLE
[Serving] Restructure voting ensemble

### PR DESCRIPTION
**Pull request by zilbermanor**

### CODE REVIEW

**Comment by yaronha** [comment]
@zilbermanor can u add a test for it (see under /tests/serving)

File: mlrun/serving/routers.py [hunk 1]
```diff
@@ -211,20 +231,68 @@ class VotingEnsemble(BaseModelRouter):
...
+        ```
+
+        How to extend the VotingEnsemble
+        --------------------------------
+        The VotingEnsemble applies its logic using the `logic(predictions)` function.
+        The `logic()` function receives an array of (# samples, # predictors) which you
+        can then use to apply whatever logic you may need.
+
+        If we use this `VotingEnsemble` as an example, the `logic()` function tries to figure
+        out weather you are trying to do a **classification** or a **regression** prediction by
```
Comment (Hedingber) [inline mlrun/serving/routers.py:262]:
```suggestion
        out whether you are trying to do a **classification** or a **regression** prediction by
```

File: mlrun/serving/routers.py [hunk 2]
```diff
@@ -265,39 +335,63 @@ def _resolve_route(self, body, urlpath):
...
+                # Path =  <prefix>/<segment>
                 # Are we looking at a router level operation?
-                if segments[0] in ["infer", "predict", "explain"]:
-                    # We are given an operation
+                try:
+                    operation = OperationTypes(segments[0])
                     self.log_router = True
-                    return self.name, None, segments[0]
-            model = segments[0]
+                    return self.name, None, operation
```
Comment (Hedingber) [inline mlrun/serving/routers.py:351]: The line you want to "catch" exceptions to is only the first one, so please change to:
 you're expecting to fail
```
                try:
                    operation = OperationTypes(segments[0])
                # Unrecognized operation was given, probably a model name
                except ValueError:
                    model = segments[0]
                else:
                    self.log_router = True
                    return self.name, None, operation

**Review by Hedingber — APPROVED** [review]
Looks good, approved, fix the two minor comments and I'll merge

**Pull request merged**